### DISCUSSION TITLE
Domain-specific rewrites

### DISCUSSION THREAD

**Discussion by kripod** [vercel/vercel#5291] [Ideas] [answered]
Subdomains and URL rewrites are great features on their own, but (dis)allowing [rewrites](https://vercel.com/docs/configuration#project/rewrites) based on aliases would open up new possibilities. Similarly to [this example](https://github.com/dcangulo/nextjs-subdomain-example), someone may want to serve the contents of each subdomain from a dedicated page, e.g. `/custom-site`, while leaving other routes intact.

An example config for the routing described above may look like:

```json
{
  "alias": ["my-domain.com", "*.my-domain.com"],
  "rewrites": [
    {
      "forAlias": ["*.my-domain.com"],
      "source": "/:match*",
      "destination": "/custom-site/:match*"
    }
  ]
}
```

No multiple deployments would be required. Also, subdomains wouldn’t inherit the contents of root domains (which would become an issue while serving assets).

**Comment by paulogdm** [comment, marked as answer]
"alias" is deprecated and **should not** be used. We are indeed exploring a way to make this type of rewrite possible.

**Reply by dbauszus-glx** [reply]
Is this already possible?

We have incorporated a blog into our website hosted with vercel. We would like to keep links for blog articles alive by using a rewrite for the blog domain which is aliased to the website domain.

e.g. blog.geolytix.net/* => geolytix.com/blog/*

**Comment by kripod** [comment]
@paulogdm Thanks for your quick answer! I'm really excited that you're already considering this functionality.

**Comment by mdathersajjad** [comment]
If anyone is trying to figure this out. Next js has added a new feature in rewrites check this out 
https://nextjs.org/docs/api-reference/next.config.js/rewrites#header-cookie-and-query-matching
